lib/src/components/NxToggle/__tests__/NxToggle.test.tsx

```diff
@@ -5,116 +5,97 @@
-  });
+  it('renders an input element with role switch with the correct attributes and classname', function() {
+    render(<NxToggle {...simpleProps}></NxToggle>);
 
-  it('adds the tm-checked class if isChecked is true, and the tm-unchecked class if it is false', function() {
-    expect(getShallow()).toHaveClassName('tm-unchecked');
-    expect(getShallow()).not.toHaveClassName('tm-checked');
+    const checkbox = screen.getByRole('switch');
 
-    expect(getShallow({ isChecked: true })).not.toHaveClassName('tm-unchecked');
-    expect(getShallow({ isChecked: true })).toHaveClassName('tm-checked');
+    expect(checkbox).toHaveAttribute('type', 'checkbox');
+    expect(checkbox).toHaveAttribute('id', 'toggle-id');
+    expect(checkbox).toHaveClass('nx-toggle__input');
   });
 
-  it('renders children nodes within an nx-toggle__content <span>', function() {
-    const render = shallow(
-      <NxToggle { ...simpleProps }>
-        <div className="bar"></div>
-      </NxToggle>
-    );
+  it('adds classes specified with the className prop', function() {
+    const { container } = render(<NxToggle {...simpleProps} className="foo"></NxToggle>);
 
-    expect(render).toContainMatchingElement('label .nx-toggle__content .bar');
-  });
+    const label = container.querySelector('label');
 
-  it('disables the input iff disabled is set', function() {
-    expect(getShallow().find('input')).toHaveProp('disabled', false);
-    expect(getShallow({ disabled: false }).find('input')).toHaveProp('disabled', false);
-    expect(getShallow({ disabled: true }).find('input')).toHaveProp('disabled', true);
+    expect(label).toHaveClass('foo');
+    expect(label).toHaveClass('nx-toggle');
   });
 
-  it('sets the input to checked per the value of isChecked', function() {
-    expect(getShallow({ isChecked: false }).find('input')).toHaveProp('checked', false);
-    expect(getShallow({ isChecked: true }).find('input')).toHaveProp('checked', true);
-  });
+  it('calls its onChange prop when the input is clicked', async function() {
+    const user = userEvent.setup();
+
+    const onChange = jest.fn();
 
-  it('calls its onChange prop when the input fires a change event', function() {
-    const onChange = jest.fn(),
-        component = getShallow({ onChange });
+    render(<NxToggle {...simpleProps} onChange={onChange}></NxToggle>);
+
+    const checkbox = screen.getByRole('switch');
 
     expect(onChange).not.toHaveBeenCalled();
-    component.find('input').simulate('change');
-    expect(onChange).toHaveBeenCalled();
-  });
 
-  it('sets the input as readonly if there is no onChange handler', function() {
-    expect(getShallow().find('input')).toHaveProp('readOnly', false);
-    expect(getShallow({ onChange: undefined }).find('input')).toHaveProp('readOnly', true);
-    expect(getShallow({ onChange: null }).find('input')).toHaveProp('readOnly', true);
-  });
+    await user.click(checkbox);
```

Comment:
It's really when anywhere in the label is clicked.  This is especially important since the `<input type="checkbox">` is invisible.

Reply:
> It's really when anywhere in the label is clicked. This is especially important since the `<input type="checkbox">` is invisible.

Updated the click test here: 5617387e0a013b6eb0a477e5b48c09008e7305e3